### DISCUSSION TITLE
Why does my software brightness gets reset to max when also using f.lux?

### DISCUSSION THREAD

**Discussion by thedeagler** [MonitorControl/MonitorControl#647] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

The software brightness seems to work for a second before getting reset to max brightness. I've noticed that if I disable flux, the problem goes away. Not sure if it's possible to have two apps controlling monitor brightness, but my guess is that flux is updating the colors every second, and perhaps it's not possible to override.

### Steps to reproduce

1. Enable flux
2. Change brightness using software dimming
3. Observe momentary dimming ~1s before the monitor returns to original brightness.

### Expected behavior

Dimming stays at the set level.

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.6 big sur
- Mac model: imac pro
- MonitorControl version: 3.1.1 b3096
- Monitor(s): Lenovo Y27q-20
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment, marked as answer]
Yes, this is a problem, MonitorControl and f.lux can't work together as both are trying to manipulate the gamma tables and cancel each other out, f.lux being the more aggressive one. I recommend using Night Shift instead. For use cases when Night Shift is not an option (for example Night Shift is disabled on 'TV as a monitor' scenarios), we can add a feature for color temperature presets + a timer when to change presets - replicating Night Shift functionality. If the other devs approve this as scope, we can make it happen.

**Update:** the new v4.0.0 version has an alternate dimming mode (shade) which can be activated by clicking `Avoid gamma table manipulation`'` in Preferences/Displays (the option is visible only when Advanced settings are enabled). This has some drawbacks (for example the mouse cursor is not dimmed), but it is compatible with flux.

Also, v4.0.0 version _(note: v4.0.0-beta2 still does not have this feature)_ will detect if an external app interferes with MonitorControl and will offer to turn off gamma table manipulation. Every time a brightness a change is initiated by the user, we check the gamma table - if at least three times the current gamma table peak value mismatches the expected value, an alert will be triggered notifying the user about the problem.

**Comment by waydabber** [comment]
I'll this move over to discussion as this f.lux issue comes up from time to time and it is educational.

**Comment by thedeagler** [comment]
Ah, this makes sense. It would be cool if monitor control and flux could work together, but I'll use nightshift for now. Thanks for answering the question!

**Comment by waydabber** [comment]
Note: The new v4.0.0 has an alternate dimming mode (shade) which can be activated by clicking 'Avoid gamma table manipulation' in Preferences/Displays. This has some drawbacks, but is complatible with flux.

**Comment by waydabber** [comment]
The upcoming v4.0.0 version will detect if an external app interferes with MonitorControl and will offer to turn off gamma table manipulation (every time a brightness a change is initiated by the user, we check the gamma table - if at least three times the current gamma table peak value mismatches the expected value, an alert will be triggered notifying the user about the problem).
